tests/unit/s3/test_connection.py

```diff
@@ -48,6 +48,18 @@ def test_switched(self):
         )
 
 
+class TestAnon(MockServiceWithConfigTestCase):
+    connection_class = S3Connection
+
+    def test_generate_url(self):
+        conn = self.connection_class(
+            anon=True,
```

Comment:
Is there another test for that exercises that the non-anonymous case generates a URL as expected?